src/main/java/mate/jdbc/lib/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,136 @@
+package mate.jdbc.lib;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import mate.jdbc.DataProcessingException;
+import mate.jdbc.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    static {
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+        } catch (ClassNotFoundException e) {
+            throw new DataProcessingException("can't create a jdbc driver ", e);
+        }
+    }
+
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertFormatRequest = "INSERT INTO Manufacturers(name, country)"
+                + " VALUES(?,?)";
+        int nameIndex = 1;
+        int countryIndex = 2;
+        int idIndex = 1;
+        try (Connection connection = getConnection(); PreparedStatement creationStatement =
+                connection.prepareStatement(insertFormatRequest, Statement.RETURN_GENERATED_KEYS)) {
+            creationStatement.setString(nameIndex, manufacturer.getName());
+            creationStatement.setString(countryIndex, manufacturer.getCountry());
+            creationStatement.executeUpdate();
+            ResultSet generatedKeys = creationStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(idIndex, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("can't insert manufacturer to DB ", e);
+        }
+        return manufacturer;
+    }
+
+    public Optional<Manufacturer> get(Long id) {
+        String selectRequest = "SELECT * FROM Manufacturers WHERE id = ? AND is_deleted = FALSE";
+        int idPosition = 1;
+        int namePosition = 2;
+        int countryPosition = 3;
+        Manufacturer manufacturer = new Manufacturer();
+        try (Connection connection = getConnection(); PreparedStatement selectStatement =
+                connection.prepareStatement(selectRequest)) {
+            selectStatement.setLong(idPosition, id);
+            selectStatement.executeQuery();
+            ResultSet generatedKeys = selectStatement.executeQuery();
+            manufacturer.setId(id);
+            if (generatedKeys.next()) {
+                String name = generatedKeys.getObject(namePosition, String.class);
+                String country = generatedKeys.getObject(countryPosition, String.class);
+                manufacturer.setName(name);
+                manufacturer.setCountry(country);
```

Comment:
```suggestion
                String name = generatedKeys.getObject(namePosition, String.class);
                String country = generatedKeys.getObject(countryPosition, String.class);
                manufacturer.setName(name);
                manufacturer.setCountry(country);
```

create method for parsing manufacturer from result set